codex/sales.nim

```diff
@@ -117,15 +148,39 @@ proc processSlot(sales: Sales, item: SlotQueueItem, done: Future[void]) =
     none StorageRequest
   )
 
-  agent.context.onCleanUp = proc {.async.} =
-    await sales.remove(agent)
+  agent.onCleanUp = proc {.async.} =
+    await sales.cleanUp(agent, done)
 
   agent.context.onFilled = some proc(request: StorageRequest, slotIndex: UInt256) =
       sales.filled(done)
 
   agent.start(SalePreparing())
   sales.agents.add agent
 
+proc deleteInactiveReservations(sales: Sales, activeSlots: seq[Slot]) {.async.} =
+  let reservations = sales.context.reservations
+  without reservs =? await reservations.all(Reservation):
+    info "no unused reservations found for deletion"
+
+  let unused = reservs.filter(r => (
+    let slotId = slotId(r.requestId, r.slotIndex)
+    not activeSlots.any(slot => slot.id == slotId)
+  ))
+  info "found unused reservations for deletion", unused = unused.len
+
+  for reservation in unused:
+
+    logScope:
+      reservationId = reservation.id
+      availabilityId = reservation.availabilityId
+
+    if err =? (await reservations.deleteReservation(
+      reservation.id, reservation.availabilityId
+    )).errorOption:
+      error "failed to delete unused reservation", error = err.msg
+    else:
+      trace "deleted unused reservation"
```

Comment:
This will log this message for every reservation deleted, right? Maybe only one would be sufficient? 😅